src/Compiler/Optimize/Optimizer.fs

```diff
@@ -3107,9 +3124,9 @@ and OptimizeVal cenv env expr (v: ValRef, m) =
            let e, einfo = OptimizeExpr cenv env e 
            e, AddValEqualityInfo g m v einfo 
 
-    | None -> 
-       if v.MustInline then
-           error(Error(FSComp.SR.optFailedToInlineValue(v.DisplayName), m))
+    | None ->
+       if v.ShouldInline then
+           warning(Error(FSComp.SR.optFailedToInlineValue(v.DisplayName), m))
```

Comment:
See other point in test expectations: I think this change from `error` to `warning` would deserve more explanation, since it will effect how e.g. incorrect `inline rec` gets reported.

Reply:
Replaced comment with:
            // If the lambdo expression uses uses a private binding then don't inline